Automattic/pocket-casts-android
# End of Year: Check user eligibility and listening history

| 📘 Project: #410 |
|:---:|

iOS PR: https://github.com/Automattic/pocket-casts-ios/pull/433

## Description

This PR 
* Checks if a user is eligible to see their EoY stats (i.e. they listened to at least one episode for more than 30 minutes this year)
* Checks user Listening History, if their history is not complete, call a placeholder sync method.

Note that the "listening history" sync endpoint is not yet integrated.

#### Checking Listening History

Right now, the app assumes that the user has the full listening history based on:

1. If their oldest item on Listening History is from a previous year
2. If the above is false check if their total number of episodes in Listening History is smaller or equal to 100 (the server limit)

## Testing Instructions

1. Set `END_OF_YEAR_ENABLED` feature flag to `true` in `base.gradle`
2. Do a clean install of the app and run it
3. ✅ Check that nothing happens, no modal is shown and there's no card under Profile
4. Login to account in which you have a Listening History with a few episodes
5. ✅ Check that as soon as sync completes, the modal is shown and the card is now visible on Profile

## Checklist
- If this is a user-facing change, I have added an entry in CHANGELOG.md
- All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
- Any jetpack compose components I added or changed are covered by compose previews
 
#### I have tested any UI changes...

- with different themes
- with a landscape orientation
- with the device set to have a large display and font size
- for accessibility with TalkBack
